src/admin/Guardian.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.21;
+
+import {Root} from "src/Root.sol";
+
+interface SafeLike {
+    function getOwners() external view returns (address[] memory);
+    function isOwner(address signer) external view returns (bool);
+    function getThreshold() external view returns (uint256);
+}
+
+interface RouterAggregatorLike {
+    function disputeMessageRecovery(bytes32 messageHash) external;
+}
+
+/// @title  Guardian
+/// @dev    This contract allows a Gnosis Safe to schedule and cancel new relys,
+///         and unpause the protocol through the timelock of Root. Additionally,
+///         it allows any owners of the safe to instantly pause the protocol.
+
+contract Guardian {
+    Root public immutable root;
+    SafeLike public immutable safe;
+    RouterAggregatorLike public immutable aggregator;
+
+    constructor(address root_, address safe_, address aggregator_) {
+        root = Root(root_);
+        safe = SafeLike(safe_);
+        aggregator = RouterAggregatorLike(aggregator_);
+    }
+
+    modifier onlySafe() {
+        require(msg.sender == address(safe), "Guardian/not-an-authorized-safe");
+        _;
+    }
+
+    modifier onlyOwner() {
+        require(_isSafeOwner(safe, msg.sender), "Guardian/not-an-owner-of-the-authorized-safe");
```

Comment:
I'd change this to:
```suggestion
    modifier onlySafeOrOwner() {
        require(msg.sender == address(safe) || _isSafeOwner(safe, msg.sender), "Guardian/not-an-owner-of-the-authorized-safe");
```

Why: this way, we can also use the entire guardian functionalities without an actual Gnosis Safe deployment (where `safe` is maybe another multisig, or just an EOA). This is especially helpful on testnet where there might not always be Gnosis Safe support